test/extended/setup_compat.sh

```diff
@@ -0,0 +1,354 @@
+    configure_versioned_os_server
+  else
+    configure_os_server
+  fi
+  #turn on audit logging for extended tests ... mimic what is done in util.sh configure_os_server, but don't
+  # put change there - only want this for extended tests
+  echo "[INFO] Turn on audit logging"
+  cp ${SERVER_CONFIG_DIR}/master/master-config.yaml ${SERVER_CONFIG_DIR}/master/master-config.orig2.yaml
+  openshift ex config patch ${SERVER_CONFIG_DIR}/master/master-config.orig2.yaml --patch="{\"auditConfig\": {\"enabled\": true}}"  > ${SERVER_CONFIG_DIR}/master/master-config.yaml
+
+  # Similar to above check, if the XFS volume dir mount point exists enable
+  # local storage quota in node-config.yaml so these tests can pass:
+  if [ -d "/mnt/openshift-xfs-vol-dir" ]; then
+      # The ec2 images usually have ~5Gi of space defined for the xfs vol for the registry; want to give /registry a good chunk of that
+      # to store the images created when the extended tests run
+    sed -i 's/perFSGroup: null/perFSGroup: 4480Mi/' $NODE_CONFIG_DIR/node-config.yaml
+  fi
+  echo "[INFO] Using VOLUME_DIR=${VOLUME_DIR}"
+
+  start_etcd
+  start_api_server
+  start_controllers
+
+  if [[ -z "${SKIP_NODE:-}" ]]; then
+    start_node
+  fi
+
+  export KUBECONFIG="${ADMIN_KUBECONFIG}"
+
+  install_registry
+  if [[ -z "${SKIP_NODE:-}" ]]; then
+    wait_for_registry
+    DROP_SYN_DURING_RESTART=1 CREATE_ROUTER_CERT=1 install_router
+    echo "[INFO] Creating image streams"
+    oc create -n openshift -f examples/image-streams/image-streams-centos7.json --config="${ADMIN_KUBECONFIG}"
+  fi
+  return
+}
+
+# start_etcd starts etcd for OpenShift
+function start_etcd() {
+  local sudo="${USE_SUDO:+sudo}"
+
+  echo "[INFO] Etcd logs will be at:      ${LOG_DIR}/etcd.log"
+  echo "[INFO] Test artifacts will be in: ${ARTIFACT_DIR}"
+  echo "[INFO] Volumes dir is:            ${VOLUME_DIR}"
+  echo "[INFO] Config dir is:             ${SERVER_CONFIG_DIR}"
+
+  mkdir -p ${LOG_DIR}
+
+  echo "[INFO] Starting etcd"
+  ${sudo} env "PATH=${PATH}" OPENSHIFT_ON_PANIC=crash openshift start etcd \
+    --config=${MASTER_CONFIG_DIR}/master-config.yaml &>"${LOG_DIR}/etcd.log" &
+  export ETCD_PID=$!
+
+  echo "[INFO] etcd server start at: "
+  date 
+
+  wait_for_url "${API_SCHEME}://${API_HOST}:${ETCD_PORT}/version" "etcd: " 0.25 80
+
+  echo "[INFO] etcd server health checks done at: "
+  date
+}
+
+# start_api_server starts the OpenShift API server
+function start_api_server() {
+  local sudo="${USE_SUDO:+sudo}"
+
+  echo "[INFO] Master API server logs will be at:    ${LOG_DIR}/apiserver.log"
+  echo "[INFO] Config dir is:                        ${SERVER_CONFIG_DIR}"
+  echo "[INFO] MasterIP is:                          ${MASTER_ADDR}"
+
+  mkdir -p ${LOG_DIR}
+
+  echo "[INFO] Starting OpenShift API server"
+  if [[ -n "${API_SERVER_VERSION:-}" ]]; then
+    ${sudo} docker run -e OPENSHIFT_PROFILE=web -e OPENSHIFT_ON_PANIC=crash \
```

Comment:
these complicated invocations are a little easier to understand with a more sparse style:
```sh
${sudo} docker run -e OPENSHIFT_PROFILE=web                       \
                   -e OPENSHIFT_ON_PANIC=crash                    \
                   --privileged                                   \
                   --net=host                                     \
                   -v "${MASTER_CONFIG_DIR}:${MASTER_CONFIG_DIR}" \
                   "openshift/origin:${API_SERVER_VERSION}" start master api                                \
                                                         --config="${MASTER_CONFIG_DIR}/master-config.yaml" \
                                                         > "${LOG_DIR}/apiserver.log" 2>&1 &
```